### PR TITLE
Update usage of Trademarks

### PR DESCRIPTION
**Pull request by awholegunch**
Adds info on Trademarks, adds trademark symbols, moves the trademark symbol replace directives to the common file, so they are available in all files.

Corrects some usages of trademarks. Makes the copyright string available as a replace directive, and uses it on the License page.

Staged at: [2.8.0-SNAPSHOT-2.8.0_trademarks](http://stg-web101.sw.joyent.continuuity.net/cdap/2.8.0-SNAPSHOT-2.8.0_trademarks/en/index.html)

### CODE REVIEW

File: cdap-docs/reference-manual/source/licenses/index.rst [hunk 1]
```diff
@@ -13,12 +13,61 @@ Cask Data Application Platform Licenses and Dependencies
...
+
+
+Other Products and Trademarks
+-----------------------------
+
+Other product and company names mentioned herein, within our documentation or webpages, may
+be the trademarks of their respective owners. Where the ownership or registration is known,
+the addition of appropriate symbols has been made.
+
+Apache®, Apache Hadoop, and Hadoop® are either registered trademarks or trademarks of the
```
Comment (albertshau) [inline cdap-docs/reference-manual/source/licenses/index.rst]: do we need a (r) after Apache Hadoop here?

**Comment by albertshau** [comment]
one question, lgtm

File: cdap-docs/reference-manual/source/licenses/index.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/licenses/index.rst]: Apparently not. This was copied directly from the Apache Hadoop website: 
http://www.apache.org/foundation/marks/faq/#attribution

Comment (awholegunch) [inline cdap-docs/reference-manual/source/licenses/index.rst]: Though now that I re-read it, they suggest adding hyperlinks. Which I will do...

Comment (awholegunch) [inline cdap-docs/reference-manual/source/licenses/index.rst]: Done.

**Comment by awholegunch** [comment]
Thanks. Addressed comment, merging.

**Pull request merged**